File: app/src/main/java/alexis/tvrs/rickandmortyuniverse/ui/locations/LocationAdapter.java
package alexis.tvrs.rickandmortyuniverse.ui.locations;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.ArrayList;

import alexis.tvrs.rickandmortyuniverse.R;

public class LocationAdapter extends BaseAdapter{
    private Context context;
    private ArrayList<Location> list;

    LocationAdapter(Context context, ArrayList<Location> list) {
        this.context = context;
        this.list = list;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LinearLayout layoutItem;
        LayoutInflater mInflater = LayoutInflater.from(context);

        if (convertView == null) {
            layoutItem = (LinearLayout) mInflater.inflate(R.layout.location_layout, parent, false);
        } else {
            layoutItem = (LinearLayout) convertView;
        }

        TextView nameView =layoutItem.findViewById(R.id.locationLayout_name);
        String nameText = "Name: "+list.get(position).getName();
        nameView.setText(nameText);

        TextView typeView =layoutItem.findViewById(R.id.locationLayout_type);
        String typeText = "Type: "+list.get(position).getType();
        typeView.setText(typeText);

        TextView dimensionView =layoutItem.findViewById(R.id.locationLayout_dimension);
        String dimensionText = "Dimension: "+list.get(position).getDimension();
        dimensionView.setText(dimensionText);


        return layoutItem;
    }
}

File: app/src/main/java/alexis/tvrs/rickandmortyuniverse/wiki/adapters/LocationAdapter.kt
package alexis.tvrs.rickandmortyuniverse.wiki.adapters

import alexis.tvrs.rickandmortyuniverse.R
import alexis.tvrs.rickandmortyuniverse.wiki.models.Location
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.LinearLayout
import kotlinx.android.synthetic.main.adapter_location.view.*
import java.util.*

class LocationAdapter internal constructor(private val context: Context, private val list: ArrayList<Location>) : BaseAdapter() {
    override fun getCount(): Int {
        return list.size
    }

    override fun getItem(position: Int): Any {
        return list[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val layoutItem: LinearLayout
        val mInflater = LayoutInflater.from(context)
        layoutItem = if (convertView == null) {
            mInflater.inflate(R.layout.adapter_location, parent, false) as LinearLayout
        } else {
            convertView as LinearLayout
        }
        val nameText = "Name: " + list[position].name
        layoutItem.locationLayout_name.text = nameText

        val typeText = "Type: " + list[position].type
        layoutItem.locationLayout_type.text = typeText

        val dimensionText = "Dimension: " + list[position].dimension
        layoutItem.locationLayout_dimension.text = dimensionText
        return layoutItem
    }
}